mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,662 @@
+                  name: "avoidBuggyIps"
+                  description: |
+                    If true, avoid using IPs ending in .0 or .255.
+                    This avoids buggy consumer devices mistakenly dropping IPv4 traffic for
+                    those special IP addresses.
+                  default_from_api: true
+                - !ruby/object:Api::Type::Boolean
+                  name: "manualAssign"
+                  description: |
+                    If true, prevent IP addresses from being automatically assigned.
+                  default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: vcenter
+    description: |
+      VmwareVCenterConfig specifies vCenter config for the user cluster.
+      Inherited from the admin cluster.
+    output: true
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: "resourcePool"
+        description: The name of the vCenter resource pool for the user cluster.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "datastore"
+        description: The name of the vCenter datastore for the user cluster.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "datacenter"
+        description: The name of the vCenter datacenter for the user cluster.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "cluster"
+        description: The name of the vCenter cluster for the user cluster.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "folder"
+        description: The name of the vCenter folder for the user cluster.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "caCertData"
+        description: Contains the vCenter CA certificate public key for SSL verification.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: "address"
+        description: The vCenter IP address.
+        output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: status
+    description: ResourceStatus representing detailed cluster state.
+    output: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: "errorMessage"
+        description: |
+          Human-friendly representation of the error message from the user cluster
+          controller. The error message can be temporary as the user cluster
+          controller creates a cluster or node pool. If the error message persists
+          for a longer period of time, it can be used to surface error message to
+          indicate real problems requiring user intervention.
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: "conditions"
+        description: |
+          ResourceConditions provide a standard mechanism for higher-level status reporting from user cluster controller.
+        output: true
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
```

Comment:
nit: subproperties should be `output: true` as well (we should really validate for this one)

Reply:
Corrected